### DISCUSSION TITLE
Array Operations

### DISCUSSION THREAD

**Discussion by robert-avena** [cooperrc/computational-mechanics#143] [Q&A]
Hi everyone,

I'm looking at the 02_Working_with_Python notebook in module_01 and I'm having trouble understanding the outputs of these array operations.

![Screen Shot 2022-12-28 at 1 13 44 PM](https://user-images.githubusercontent.com/121516267/209855053-c388b2c6-19a7-4d0c-90fb-40bf0d85e7e0.png)

I would expect the output to be an array like ```[1, 1, 81]``` for the first one, but the output instead has several values with weird spacing. Can anyone explain it?

**Comment by C-Doll** [comment]
the line
x_array = np.linspace(-1,1,9) isn't creating an array of [-1,1,9], it is creating an array from -1, to 1, with 9 total elements. 
![image](https://user-images.githubusercontent.com/121252243/209865603-f1a6438c-d4e6-4a8c-a123-9e263cf47d52.png)

You can then square them or take the square root.

Hope this helps,

-Chris.

**Reply by robert-avena** [reply]
Ah, yes I thought it was creating a ```[-1, 1, 9]``` array. Makes sense now, thank you!

**Comment by cooperrc** [comment]
Oooh, cool catch.

[np.linspace docs](https://numpy.org/doc/stable/reference/generated/numpy.linspace.html) state, 

```python
numpy.linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None, axis=0)
```

so if you ask for `start = -1`,  `stop = 1`, and `num = 9` you get 9 equally-spaced values from -1 to 1
